checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -3367,15 +3370,21 @@ public void analyzeObligationFulfillingLoop(
    * @param lastLoopBodyBlock last block of loop body
    * @param potentiallyFulfillingLoop loop wrapper of the loop to analyze
    * @param obligations the set of tracked obligations
+   * @param loopUpdateBlock block that updates the loop
    * @return the union of methods in the CalledMethods type of the collection element and all its
-   *     resource aliases.
+   *     resource aliases or {@code null} if the called methods is bottom
    */
   private Set<String> analyzeTypeOfCollectionElement(
       Block lastLoopBodyBlock,
       PotentiallyFulfillingLoop potentiallyFulfillingLoop,
-      Set<Obligation> obligations) {
+      Set<Obligation> obligations,
+      Block loopUpdateBlock) {
     AccumulationStore store = null;
-    if (lastLoopBodyBlock.getLastNode() == null) {
+    if (lastLoopBodyBlock.getType() == BlockType.CONDITIONAL_BLOCK) {
+      ConditionalBlock conditionalBlock = (ConditionalBlock) lastLoopBodyBlock;
+      boolean thenSuccessor = conditionalBlock.getThenSuccessor() == loopUpdateBlock;
+      store = cmAtf.getStoreAfterConditionalBlock(conditionalBlock, thenSuccessor);
+    } else if (lastLoopBodyBlock.getLastNode() == null) {
       // TODO is this really the right store? I think we need to get the then-or else store
       store = cmAtf.getStoreAfterBlock(lastLoopBodyBlock);
```

Comment:
I don't think this can happen anymore.  I would add an exception and see when this happens.